# HG changeset patch
# User Andreas Weber <andy.weber.aw@gmail.com>
# Date 1411327951 -7200
#      Sun Sep 21 21:32:31 2014 +0200
# Node ID 10c57204fd726d09bfa05fe2c8cb823a69339938
# Parent  dc51f26f0db511607af9ff1be2e5f7aec5b44855
nproc.cc: codesprint: adding tests

diff --git a/libinterp/corefcn/nproc.cc b/libinterp/corefcn/nproc.cc
--- a/libinterp/corefcn/nproc.cc
+++ b/libinterp/corefcn/nproc.cc
@@ -82,9 +82,26 @@ variable.\n\
   retval = num_processors (query);
 
   return retval;
 }
 
 /*
 ## Must always report at least 1 cpu available
 %!assert (nproc () >= 1);
+%!assert (nproc ("all") >= 1);
+%!assert (nproc ("current") >= 1);
+
+%!test
+%! c = nproc ("current");
+%! unwind_protect
+%!   old_val = getenv ("OMP_NUM_THREADS");
+%!   new_val = c + 1;
+%!   setenv ("OMP_NUM_THREADS", num2str (new_val));
+%!   assert (nproc ("overridable"), new_val);
+%! unwind_protect_cleanup
+%!   if (! isempty (old_val))
+%!     setenv ("OMP_NUM_THREADS", old_val);
+%!   endif
+%! end_unwind_protect
+
+%!error nproc ("no_valid_option");
 */
